### ISSUE TITLE
Projected configmap not interpreted

### ISSUE THREAD

**Issue by george-angel** [kubernetes-sigs/kustomize#153]
My base configuration contains:

```
...
      volumes:
        - name: prometheus-config-shared
          emptyDir: {}
        - name: prometheus-config
          configMap:
            name: prometheus-config
        - name: prometheus-alerts
          projected:
            sources:
              - configMap:
                  name: prometheus-alerts
...
```

kustomization.yaml:

```
...
configMapGenerator:
  - name: prometheus-config
    files:
      - prometheus-config.yaml
  - name: prometheus-alerts
    files:
      - system-alerts.yaml
```

`kustomize build` produces:

```
      volumes:
      - emptyDir: {}
        name: prometheus-config-shared
      - configMap:
          name: prometheus-config-ght7d485tk
        name: prometheus-config
      - name: prometheus-alerts
        projected:
          sources:
          - configMap:
              name: prometheus-alerts
```

And so my deployment fails:

```
MountVolume.SetUp failed for volume "prometheus-alerts" : configmaps "prometheus-alerts" not found
```

```
NAME                                     DATA      AGE
configmap/prometheus-alerts-gtkkhc6md2   1         2d
```

**Comment by robbyt** [comment]
I have the same issue, but I'm not even using the `configMapGenerator`.  Hashes are not being appended onto the configMap names under the projected configMap volume.

```
      volumes:
      - name: nginx-config
        projected:
          sources:
          - configMap:
              name: nginx-v1
          - configMap:
              name: nginx-main-v1
          - configMap:
              name: nginx-public-default-redirect-v1
          - configMap:
              name: nginx-members-v1
```

**Comment by robbyt** [comment]
I'd like to send a PR for this, but I can't figure out how to add a matching test for it. Anyhow, here is the fix for this:

```diff
diff --git a/pkg/transformers/namereferenceconfig.go b/pkg/transformers/namereferenceconfig.go
index b34380b..0f07250 100644
--- a/pkg/transformers/namereferenceconfig.go
+++ b/pkg/transformers/namereferenceconfig.go
@@ -146,6 +146,13 @@ var defaultNameReferencePathConfigs = []ReferencePathConfig{
                                Path:               []string{"spec", "template", "spec", "initContainers", "envFrom", "configMapRef", "name"},
                                CreateIfNotPresent: false,
                        },
+                 {
+                         GroupVersionKind: &schema.GroupVersionKind{
+                                 Kind: "Deployment",
+                         },
+                         Path:               []string{"spec", "template", "spec", "volumes", "projected", "sources", "configMap", "name"},
+                         CreateIfNotPresent: false,
+                 },
                        {
                                GroupVersionKind: &schema.GroupVersionKind{
                                        Kind: "ReplicaSet",
@@ -258,6 +265,13 @@ var defaultNameReferencePathConfigs = []ReferencePathConfig{
                                Path:               []string{"spec", "template", "spec", "volumes", "configMap", "name"},
                                CreateIfNotPresent: false,
                        },
+                 {
+                         GroupVersionKind: &schema.GroupVersionKind{
+                                 Kind: "StatefulSet",
+                         },
+                         Path:               []string{"spec", "template", "spec", "volumes", "projected", "sources", "configMap", "name"},
+                         CreateIfNotPresent: false,
+                 },
                        {
                                GroupVersionKind: &schema.GroupVersionKind{
                                        Kind: "Job",

```

**Comment by Liujingfang1** [comment]
@robbyt The change looks good to me. You can add tests similar to these ones https://github.com/kubernetes-sigs/kustomize/blob/master/pkg/transformers/namereference_test.go

**Comment by george-angel** [comment]
Thanks both, I opened a PR: https://github.com/kubernetes-sigs/kustomize/pull/158

**Comment by george-angel** [comment]
Fixed in 1.0.4
